### PR TITLE
Update UserSecret E2E Tests

### LINKED ISSUE #3833: Refactor UserSecret E2E Tests
### Description

All of the user secret E2E tests are not currently validating the right behaviour. The tests just check that a specific log message is logged without actually triggering the function that utilizes user secrets to see if this works as expected E2E.

[tests](https://github.com/Azure/azure-functions-core-tools/blob/37b826f143b99cd62308b88320e71aa806f6197f/test/Azure.Functions.Cli.Tests/E2E/StartTests.cs#L646-L788)

### PR DESCRIPTION
**Pull request by liliankasem**
### Issue describing the changes in this PR

resolves #3833

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

- Added new queue helper
- Added storage emulator script
- Refactored user secret tests to make sure a function can still execute successfully with user secrets

> I refactored a lot of the CliTester/StartTests but the PR was getting too big and complicated so I'm splitting this out into multiple PRs.

Chain of PRs:
1. #3847
2. #3850 
3. #3846

### CODE REVIEW

File: test/Azure.Functions.Cli.Tests/E2E/StartTests.cs [hunk 1]
```diff
@@ -640,7 +640,7 @@ await CliTester.Run(new RunConfiguration
             }, _output);
         }
 
-        [Theory(Skip = "https://github.com/Azure/azure-functions-core-tools/issues/3644")]
+        [Theory]
```
Comment (mattchenderson) [inline test/Azure.Functions.Cli.Tests/E2E/StartTests.cs:643]: Should completion of this PR also resolve #3644 ?

**Review by mattchenderson — COMMENTED** [review]
From a quick mobile review this one is looking good. Very excited to move things over to the emulator! I did notice that there's still a connection string in the `userSecrets` dictionary in `CliTester.Run()` (right below a removed line). Should that also be moved to the emulator? I guess that might have some side effects for other tests, so maybe that's for one of the split up PRs mentioned in the description.

File: test/Azure.Functions.Cli.Tests/E2E/StartTests.cs (hunk 1 above)
Comment (liliankasem) [reply test/Azure.Functions.Cli.Tests/E2E/StartTests.cs:643]: I don't think so, I believe that issue is a bit if a red herring. User secrets does work for connection strings and other settings. However, it does not work for setting "FUNCTIONS_WORKER_RUNTIME" and I believe that is a different core tools issue to the loading and utilization of user secrets.

**Review by surgupta-msft — APPROVED** [review]

**Comment by surgupta-msft** [comment]
The changes look good to me. Will wait for pipelines to be green in case that requires additional significant changes.

**Review by surgupta-msft — APPROVED** [review]

**Review by satvu — APPROVED** [review]

**Pull request merged**